src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +15,47 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(actor);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can`t add actor entity " + actor, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return actor;
     }
 
     @Override
     public Optional<Actor> get(Long id) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            Actor aquiredActor = session.get(Actor.class, id);
+            transaction.commit();
+            return Optional.ofNullable(aquiredActor);
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Error occurred "
+                    + "while getting actor by id: " + id
+                    + "\nConnection failed", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
```

Comment:
Common mistakes - "Use try with finally, where you use transaction and try with resources with Read operations." 
U don`t need transactions here at all, so just open session it try-with-resources